File: hashes/hash_table.rb
class HashTable
  def initialize
    @table = []
  end

  def hash_value(str)
    str[0].ord + str[1].ord
  end

  def lookup(str)
    @table[hash_value(str)].to_a.include?(str)
  end

  def store(str)
    bucket = @table[hash_value(str)]
    @table[hash_value(str)] = bucket.to_a + [str]
  end
end

File: hashes/hash_table.py
class HashTable:
  def __init__(self):
    self.table = [None] * 10000

  def store(self, string):
    hash_value = HashableString(string).hash_value()

    if self.table[hash_value] is None:
      self.table[hash_value] = [string]
    else:
      self.table[hash_value].append(string)

  def lookup(self, string):
    bucket = self.table[HashableString(string).hash_value()]

    return bucket is not None and (string in bucket)

class HashableString(str):
  def hash_value(self):
    return ord(self[0]) + ord(self[1])